osidb/models.py

```diff
@@ -1788,6 +1804,98 @@ class Meta:
         )
 
 
+class FlawReferenceManager(ACLMixinManager, TrackingMixinManager):
+    """flawreference manager"""
+
+    @staticmethod
+    def create_flawreference(flaw, url, **extra_fields):
+        """return a new flawreference or update an existing flawreference without saving"""
+        try:
+            flawreference = FlawReference.objects.get(flaw=flaw, url=url)
+            for attr, value in extra_fields.items():
+                setattr(flawreference, attr, value)
+            return flawreference
+
+        except ObjectDoesNotExist:
+            return FlawReference(flaw=flaw, url=url, **extra_fields)
+
+
+class FlawReference(TrackingMixin, ACLMixin, AlertMixin):
+    """Model representing flaw references"""
+
+    class FlawReferenceType(models.TextChoices):
+        """
+        Allowable references:
+
+        ARTICLE:
+            A link to a Security Bulletin or Knowledge Base Article specifically
+            discussing this flaw on the Customer Portal. It always begins with
+            "https://accesss.redhat.com/". It must be a Security Bulletin
+            for Major Incidents. More general articles like hardening should be
+            linked instead in EXTERNAL REFERENCES.
+
+        EXTERNAL_REFERENCES:
+            URL links to other trustworthy sources of information about this
+            vulnerability. A link should not point to a missing resource.
+            Since these links are displayed on the CVE page of the flaw, we only
+            want to include respectable sources (such as upstream advisories,
+            analysis of security researches, etc.).
+        """
+
+        ARTICLE = "ARTICLE"
+        EXTERNAL_REFERENCES = "EXTERNAL_REFERENCES"
+
+    # internal primary key
+    uuid = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    type = models.CharField(
+        choices=FlawReferenceType.choices,
+        default=FlawReferenceType.EXTERNAL_REFERENCES,
+        max_length=50,
+    )
+
+    url = models.URLField(max_length=2048)
+
+    description = models.TextField(blank=True)
+
+    # one flaw can have many references
+    flaw = models.ForeignKey(Flaw, on_delete=models.CASCADE, related_name="references")
+
+    objects = FlawReferenceManager()
+
+    class Meta:
+        """define meta"""
+
+        unique_together = ["flaw", "url"]
+
+    def _validate_article_link(self):
+        """
+        Checks that an article link begins with https://accesss.redhat.com/.
+        """
+        if self.type == self.FlawReferenceType.ARTICLE and not self.url.startswith(
+            "https://accesss.redhat.com/"
```

Comment:
is it a typo or is it actually `accesss` with 3 s?

Reply:
Just a typo, thanks!